airbyte-integrations/connectors/source-google-search-console/source_google_search_console/streams.py

```diff
@@ -0,0 +1,261 @@
+from source_google_search_console.service_account_authenticator import ServiceAccountAuthenticator
+
+BASE_URL = "https://www.googleapis.com/webmasters/v3"
+GOOGLE_TOKEN_URI = "https://oauth2.googleapis.com/token"
+ROW_LIMIT = 25000
+CLIENT_AUTH = "Client"
+
+
+class GoogleSearchConsole(HttpStream, ABC):
+    url_base = BASE_URL
+    primary_key = None
+    data_field = ""
+
+    def __init__(
+        self,
+        auth_type: str,
+        site_urls: list,
+        start_date: str,
+        end_date: str,
+        client_id: str = None,
+        client_secret: str = None,
+        refresh_token: str = None,
+        service_account_info: str = None,
+    ):
+        super().__init__()
+        self._auth_type = auth_type
+        self._site_urls = self.sanitize_urls_list(site_urls)
+        self._start_date = start_date
+        self._end_date = end_date
+
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._refresh_token = refresh_token
+
+        self._service_account_info = service_account_info
+
+    @staticmethod
+    def sanitize_urls_list(site_urls: list) -> List[str]:
+        return list(map(quote_plus, site_urls))
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        for site_url in self._site_urls:
+            yield {"site_url": site_url}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        if not self.data_field:
+            yield response.json()
+
+        else:
+            records = response.json().get(self.data_field) or []
+            for record in records:
+                yield record
+
+    @property
+    def authenticator(self) -> Union[HttpAuthenticator, None]:
+        if self._auth_type == CLIENT_AUTH:
+            return Oauth2Authenticator(
+                token_refresh_endpoint=GOOGLE_TOKEN_URI,
+                client_secret=self._client_secret,
+                client_id=self._client_id,
+                refresh_token=self._refresh_token,
+            )
+
+        else:
+            info = json.loads(self._service_account_info)
+            return ServiceAccountAuthenticator(service_account_info=info)
+
+
+class Sites(GoogleSearchConsole):
+    """
+    API docs: https://developers.google.com/webmaster-tools/search-console-api-original/v3/sites
+    """
+
+    def path(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> str:
+        return f"/sites/{stream_slice.get('site_url')}"
+
+
+class Sitemaps(GoogleSearchConsole):
+    """
+    API docs: https://developers.google.com/webmaster-tools/search-console-api-original/v3/sitemaps
```

Comment:
thanks for including API docs links, makes it much easier to see what's going on!